src/signal.rs

```diff
@@ -822,6 +839,69 @@ pub fn gen_mut<G, F>(gen_mut: G) -> GenMut<G, F>
 }
 
 
+/// A signal that maps one set of frames to another
+///
+/// # Example
+///
+/// ```rust
+/// extern crate sample;
+///
+/// use sample::{signal, Signal};
+///
+/// fn main() {
+///     let frames = signal::gen(|| [0.5]);
+///     let mut mapper = signal::map(frames, |f| [f[0], 0.25]);
+///     assert_eq!(mapper.next(), [0.5, 0.25]);
+///     assert_eq!(mapper.next(), [0.5, 0.25]);
+///     assert_eq!(mapper.next(), [0.5, 0.25]);
+/// }
+/// ```
+pub fn map<M, S, Fout>(signal: S, map: M) -> Map<M, S, Fout>
+    where M: FnMut(<S as Signal>::Frame) -> Fout,
+          S: Signal,
+          Fout: Frame,
+{
+    Map {
+        map: map,
+        signal: signal,
+        frames: core::marker::PhantomData,
+    }
+}
+
+
+/// A signal that maps one set of frames to another
+///
+/// # Example
+///
+/// ```rust
+/// extern crate sample;
+///
+/// use sample::{signal, Signal};
+///
+/// fn main() {
+///     let frames = signal::gen(|| [0.5]);
+///     let more_frames = signal::gen(|| [0.25]);
+///     let mut mapper = signal::zip_map(frames, more_frames, |(f, o)| [f[0], o[0]]);
+///     assert_eq!(mapper.next(), [0.5, 0.25]);
+///     assert_eq!(mapper.next(), [0.5, 0.25]);
+///     assert_eq!(mapper.next(), [0.5, 0.25]);
+/// }
+/// ```
+pub fn zip_map<M, S, O, Fout>(this: S, other: O, map: M) -> ZipMap<M, S, O, Fout>
+    where M: FnMut((<S as Signal>::Frame, <O as Signal>::Frame)) -> Fout,
```

Comment:
I don't think there's any need for the extra pair of parenthesis here - should be fine to take two arguments instead

Reply:
Also I think I'd prefer if the type param was just `F` instead of `Fout`. I think the distinction between input and output frames types should still be obvious, as the input frame types must be `S` and `O`'s associated `Frame` types.